plugins/out_s3/s3_multipart.c

```diff
@@ -334,7 +334,7 @@ static int complete_multipart_upload_payload(struct flb_s3 *ctx,
     int offset = 0;
     flb_sds_t etag;
     size_t size = COMPLETE_MULTIPART_UPLOAD_BASE_LEN;
-    char part_num[7];
+    char part_num[11];
```

Comment:
Are we not limited to numbers 1..10000 per docs - https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPart.html
> Part numbers can be any number from 1 to 10,000, inclusive.

Reply:
AWS might reject the request if the part number is higher than 10000 but there's an `sprintf` call which writes whichever 32 bit signed integer it gets in the part number field into that buffer and considering how memory alignment works I'm not worried about spending 8 more bytes if that means that if there is such a fringe case we get a nice error rather than a stack overflow.